test_utils/src/lib.rs

```diff
@@ -314,10 +324,23 @@ impl WorkerToWorker for WorkerToWorkerMockServer {
     ) -> Result<anemo::Response<()>, anemo::rpc::Status> {
         let message = request.into_body();
 
-        self.sender.send(message).await.unwrap();
+        self.msg_sender.send(message).await.unwrap();
 
         Ok(anemo::Response::new(()))
     }
+    async fn request_batches(
+        &self,
+        request: anemo::Request<WorkerBatchRequest>,
+    ) -> Result<anemo::Response<WorkerBatchResponse>, anemo::rpc::Status> {
+        let message = request.into_body();
+
+        self.batch_request_sender.send(message).await.unwrap();
+
+        // For testing stub, just always reply with no batches.
+        Ok(anemo::Response::new(WorkerBatchResponse {
+            batches: vec![],
+        }))
+    }
```

Comment:
With this change to request/response pairs we'll probably end up needing to have more tests for the actual handlers themselves since they're now expected to actually respond with something.

Reply:
Agreed on more tests for the handlers. I think a good pattern for the mock servers is also TBD. Do you have thoughts on this? Curious an interface similar to gMock extensions for gRPC is possible here - is there any Rust mocking library we use?

For what test client code looks like in gRPC/C++, see the bottom of
https://grpc.github.io/grpc/cpp/md_doc_unit_testing.html

It's very intuitive IMO